tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -174,4 +176,297 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+		7. Delete the SC created in step 2
+		8. Deleted the SPBM policy created in step 1
+	*/
+	ginkgo.It("[csi-block-vanilla] Verify large EZT volume creation which takes longer than vpxd timeout", func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sharedvmfsURL := os.Getenv(envSharedVMFSDatastoreURL)
+		if sharedvmfsURL == "" {
+			ginkgo.Skip(fmt.Sprintf("Env %v is missing", envSharedVMFSDatastoreURL))
+		}
+
+		largeSize := os.Getenv(envDiskSizeLarge)
+		if largeSize == "" {
+			largeSize = diskSizeLarge
+		}
+
+		scParameters := make(map[string]string)
+
+		rand.Seed(time.Now().UnixNano())
+		suffix := fmt.Sprintf("-%v-%v", time.Now().UnixNano(), rand.Intn(10000))
+		categoryName := "category" + suffix
+		tagName := "tag" + suffix
+
+		catID, tagID := createCategoryNTag(ctx, categoryName, tagName)
+		defer func() {
+			deleteCategoryNTag(ctx, catID, tagID)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfsURL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfsURL)
+		}()
+
+		ginkgo.By("Create a SPBM policy with EZT volume allocation")
+		policyID, policyName := createVmfsStoragePolicy(
+			ctx, pc, eztAllocType, map[string]string{categoryName: tagName})
+		defer func() {
+			deleteStoragePolicy(ctx, pc, policyID)
+		}()
+
+		vpxdTimeoutSecs := 90
+		setVpxdTaskTimeout(vpxdTimeoutSecs)
+		defer func() {
+			setVpxdTaskTimeout(0)
+		}()
+
+		ginkgo.By("Create SC using policy created in step 1")
+		ginkgo.By("Create a large PVC using SC created in step 2, this should take more than vpxd task timeout")
+		scParameters[scParamStoragePolicyName] = policyName
+		storageclass, pvclaim, err := createPVCAndStorageClass(client,
+			namespace, nil, scParameters, largeSize, nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			ginkgo.By("Delete the SC created in step 2")
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		start := time.Now()
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(
+			client, []*v1.PersistentVolumeClaim{pvclaim}, framework.ClaimProvisionTimeout)
+		elapsed := time.Since(start)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify PVC creation took longer than vpxd timeout")
+		gomega.Expect(elapsed > time.Second*time.Duration(vpxdTimeoutSecs)).To(
+			gomega.BeTrue(), "PVC creation was faster than vpxd timeout")
+
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		volumeID := pvs[0].Spec.CSI.VolumeHandle
+		storagePolicyMatches, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyName)
+		e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(storagePolicyMatches).To(gomega.BeTrue(), "storage policy verification failed")
+
```

Comment:
Can we also check here if volume has been created on the expected datastore?

Reply:
done